appinventor/blocklyeditor/src/blocks/utilities.js

```diff
@@ -55,6 +61,24 @@ Blockly.Blocks.Utilities.YailTypeToBlocklyType = function(yail,inputOrOutput) {
     }
 };
 
+Blockly.Blocks.Utilities.helperKeyToBlocklyType = function(helperKey, block) {
+  if (!helperKey) {
+    return null;
+  }
+  var utils = Blockly.Blocks.Utilities;
+  switch (helperKey.type) {
+    case "OPTION_LIST":
+      return utils.optionListKeyToBlocklyType(helperKey.key, block);
+  }
+  return null;
+}
+
+Blockly.Blocks.Utilities.optionListKeyToBlocklyType = function(key, block) {
+  var optionList = block.getTopWorkspace().getComponentDatabase()
+      .getOptionList(key);
+  return optionList.className + 'Enum';
```

Comment:
Bad name mangling solution. But it seems to be the best solution we have currently -- Lyn